servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp

```diff
@@ -1258,6 +1267,24 @@ void RenderForwardClustered::_fill_render_list(RenderListType p_render_list, con
 					inst->gi_offset_cache = 0xFFFFFFFF;
 				}
 			}
+
+#ifdef DEBUG_ENABLED
+			if (unlikely(get_debug_draw_mode() == RS::VIEWPORT_DEBUG_DRAW_UV2_TEXEL_DENSITY)) {
+				if (inst->data->base_type == RS::INSTANCE_MESH && inst->data->use_baked_light) {
+					Size2 lightmap_size = mesh_storage->mesh_get_lightmap_size_hint(inst->data->base) * inst->data->lightmap_scale;
+
+					if (inst->data->lightmap_size_global_uniform_pos == -2) {
+						//not allocated, try to allocate
+						inst->data->lightmap_size_global_uniform_pos = RSG::material_storage->global_shader_uniforms_unit_variable_allocate();
+					}
+
+					RSG::material_storage->global_shader_uniforms_unit_variable_update(inst->data->lightmap_size_global_uniform_pos, lightmap_size);
```

Comment:
This shouldn't be called if ``inst->data->lightmap_size_global_uniform_pos`` is invalid
```suggestion
					if (inst->data->lightmap_size_global_uniform_pos >= 0) {
						RSG::material_storage->global_shader_uniforms_unit_variable_update(inst->data->lightmap_size_global_uniform_pos, lightmap_size);
					}
```